src/include/btree.h

```diff
@@ -93,6 +93,12 @@ typedef enum { /* Start position for eviction walk */
 /* An invalid btree file ID value. ID 0 is reserved for the metadata file. */
 #define WT_BTREE_ID_INVALID UINT32_MAX
 
+#define WT_BTREE_ID_NAMESPACE_BITS 3
```

Comment:
Why we choose three bits?

Reply:
It seemed prudent to leave a couple spare. Two bits would likely be fine as well, I'm not married to it.

Reply:
Can we choose an aligned number? May be 8?

Reply:
I think eight is too many - remember that we are taking these bits from the number of possible tables, so a 24-bit table ID leaves us with only 16M tables, which we've come close to in the wild.

Can I ask what advantage an aligned numbers gets us?

Reply:
oh, I see. We only have 32 bits. I thought aligned bits may be easier in the future for calculation and storage.

Reply:
I see where you're coming from but unless there's something concrete that makes it a worse choice, I'm inclined to leave it at three.

Reply:
Another important consideration is that the `files` array in `WT_RECOVERY` is indexed by the *namespaced* file ID (e.g., see the way `r->files[fileid]` is used in `__recovery_setup_file`). Note that this needs to be namespaced, so that we don't get collisions on the follower between file IDs created locally and file IDs created on the leader.

By using 3 bits, we are already making that array 8-times larger; we shouldn't be making it any larger than that. (As such I'm even wondering if we should go down to 2 bits, so that it's only 4-times larger.)